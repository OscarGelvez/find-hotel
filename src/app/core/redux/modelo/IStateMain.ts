export interface IStateMain {
  isLoading: boolean;
}
