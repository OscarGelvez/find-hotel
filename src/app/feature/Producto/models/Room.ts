export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
