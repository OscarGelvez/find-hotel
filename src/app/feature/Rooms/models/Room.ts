export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
