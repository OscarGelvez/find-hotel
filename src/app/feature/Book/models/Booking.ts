import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
