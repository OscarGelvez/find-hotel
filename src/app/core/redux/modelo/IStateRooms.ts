import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IStateRooms {
  listRooms: IRoom[];
  selectedRoomId: number;
}
