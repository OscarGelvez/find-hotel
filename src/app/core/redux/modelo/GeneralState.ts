import { IStateBookings } from './IStateBookings';
import { IStateRooms } from './IStateRooms';
import { IStateMain } from './IStateMain';

export interface Istate {
  rooms: IStateRooms;
  bookings: IStateBookings;
  main: IStateMain;
}
