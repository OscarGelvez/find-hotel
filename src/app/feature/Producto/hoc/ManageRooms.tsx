import {
  listingRoomsAsync,
} from 'app/core/redux/acciones/productos/ProductosAcciones';
import { Istate } from 'app/core/redux/modelo/GeneralState';
import { Rooms } from '../containers/Rooms';
import { connect } from 'react-redux';

const mapStateToProps = (state: Istate) => {
  return {listRooms : state.rooms.listRooms};
};

export const ManageRooms = connect(mapStateToProps, {
  listAvailableRooms: listingRoomsAsync,
})(Rooms);
