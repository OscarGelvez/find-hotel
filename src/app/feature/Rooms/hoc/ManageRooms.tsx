import { listingRoomsAsync } from 'app/core/redux/actions/rooms/ActionsRooms';
import { Istate } from 'app/core/redux/modelo/GeneralState';
import { Rooms } from '../containers/Rooms';
import { connect } from 'react-redux';

const mapStateToProps = (state: Istate) => {
  return { listRooms: state.rooms.listRooms };
};

export const ManageRooms = connect(mapStateToProps, {
  listAvailableRooms: listingRoomsAsync,
})(Rooms);
