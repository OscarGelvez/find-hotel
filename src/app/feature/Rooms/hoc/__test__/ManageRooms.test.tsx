import { IStateMain } from 'app/core/redux/modelo/IStateMain';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { IStateRooms } from '../../../../core/redux/modelo/IStateRooms';
import { roomInfo } from '../../../../shared/utils/data';
import { IRoom } from '../../models/Room';
import { ManageRooms } from '../ManageRooms';

const mockStore = configureStore([]);

const newRoom: IRoom = roomInfo;
const listRooms: IRoom[] = [newRoom];

const rooms: IStateRooms = {
  listRooms: listRooms,
  selectedRoomId: -1,
};

const main: IStateMain = {
  isLoading: false,
};

describe('Prueba componente de orden superior HOC con redux', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rooms: rooms,
      main,
    });

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ManageRooms />
        </BrowserRouter>
      </Provider>
    );
  });

  it('Compara snapshot del componente ManageRooms renderizado', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
