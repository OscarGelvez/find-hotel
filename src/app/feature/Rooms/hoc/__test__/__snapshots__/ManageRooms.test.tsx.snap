// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente de orden superior HOC con redux Compara snapshot del componente ManageRooms renderizado 1`] = `
<div
  className="sc-AxiKw bvTGbm"
  data-testid="room"
>
  <div
    className="card-rooms"
    data-testid="find-rooms"
  >
    <div
      className="row"
    >
      <div
        className="card filter-area "
      >
        <div
          className="col"
        >
          <form
            action="#"
            data-testid="form-find-rooms"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="form-group row mb-3 mt-3"
            >
              <label
                className="col-12 col-sm-6 col-lg-2 col-form-label"
                htmlFor="fromDate"
              >
                Fecha de ingreso*
              </label>
              <div
                className="col-12 col-sm-6 col-lg-4"
              >
                <input
                  className="form-control"
                  data-testid="form-fromDate"
                  id="fromDate"
                  min="2021-04-25"
                  name="fromDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  required={true}
                  type="date"
                  value=""
                />
              </div>
              <label
                className="col-12  col-sm-6 col-lg-2 col-form-label"
                htmlFor="untilDate"
              >
                Fecha de salida*
              </label>
              <div
                className="col-12 col-sm-6 col-lg-4"
              >
                <input
                  className="form-control"
                  data-testid="form-untilDate"
                  id="untilDate"
                  min="2021-04-26"
                  name="untilDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  required={true}
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div
              className="row mb-4"
            >
              <div
                className="d-flex col-12 col-sm-6 col-lg-4 text-center"
              >
                <label
                  className="pr-2"
                  htmlFor="capacity"
                >
                  Personas
                </label>
                <select
                  aria-label=".form-select-sm example"
                  className=" filter-select"
                  data-testid="form-capacity"
                  id="capacity"
                  name="capacity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    Seleccione
                  </option>
                  <option
                    value="1"
                  >
                    1 adulto
                  </option>
                  <option
                    value="2"
                  >
                    2 adultos
                  </option>
                  <option
                    value="3"
                  >
                    3 adultos
                  </option>
                  <option
                    value="4"
                  >
                    4 adultos
                  </option>
                </select>
              </div>
              <div
                className="col-12 col-sm-6 col-lg-4 text-md-center "
              >
                <input
                  aria-label="Checkbox find room with air"
                  checked={false}
                  className="m-2"
                  data-testid="form-withAir"
                  id="withAir"
                  name="withAir"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value={false}
                />
                <label
                  className="cursor-pointer"
                  htmlFor="withAir"
                >
                  ¿Aire acondicionado?
                </label>
              </div>
              <div
                className="col-12 col-lg-4 mt-sm-2 mt-md-0"
              >
                <input
                  aria-label="Checkbox  find room with parking"
                  checked={false}
                  className="m-2"
                  data-testid="form-withParking"
                  id="withParking"
                  name="withParking"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursor-pointer"
                  htmlFor="withParking"
                >
                  ¿Parqueadero?
                </label>
              </div>
            </div>
            <div
              className="text-center mt-5"
            >
              <button
                className="sc-AxjAm cYJbNq  w-50"
                data-testid="form-button-find"
                type="submit"
              >
                Encuentralo!
              </button>
            </div>
            <div
              className="text-center mb-4 mt-1"
            >
              <button
                className="sc-AxjAm cYJbNq button-link"
                data-testid="form-button-reset"
                type="reset"
              >
                limpiar
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-rooms"
    data-testid="list-rooms"
  >
    <div
      className="row pt-5"
    >
      <div
        className="col"
      >
        <div
          className="card room-item"
          data-testid="card-information"
        >
          <div
            className="card-header room-item-header "
          >
            <h3>
              Gold Hotel 5
            </h3>
            <h4>
              8.7
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <img
                  alt="hotel room"
                  className="img-fluid"
                  height="180px"
                  src="https://static.abc.es/Media/201504/27/hotel12--644x362.jpg"
                  width="260px"
                />
              </div>
              <div
                className="col col-md-8"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-8"
                  >
                    <h5
                      className="card-title"
                    >
                      Suite full. Caroline
                    </h5>
                  </div>
                  <div
                    className="col-12 col-lg-4"
                  >
                    <a
                      className="sc-AxirZ AiLKW"
                      href="/rooms/detail"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-primary btn-book w-100"
                        data-testid="form-button-load-detail"
                        id="btn-select-detail"
                        onClick={[Function]}
                        type="button"
                      >
                        <i
                          className="far fa-calendar-check ml-3"
                        />
                         Seleccionar
                      </button>
                    </a>
                  </div>
                  <div
                    className="col-12 col-md-12"
                  >
                    <h4>
                      <i
                        className="far fa-money-bill-alt"
                      />
                       
                      154.000
                    </h4>
                  </div>
                  <div
                    className="col-12 col-md-12 d-flex"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Inicio
                       
                      <i
                        className="fas fa-calendar-alt"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        2021-04-13
                      </span>
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      -  Fin
                       
                      <i
                        className="fas fa-calendar-alt"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        2021-04-30
                      </span>
                    </h6>
                  </div>
                </div>
                <p
                  className="card-text"
                >
                  Baño privado, cama doble, con nevera, Wifi, TV.
                </p>
                <div
                  className="row"
                >
                  <div
                    className="col col-md-6"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Camas
                       
                      <i
                        className="fas fa-bed"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        1
                      </span>
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Aire
                       
                      <i
                        className="fas fa-wind"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Wifi
                       
                      <i
                        className="fas fa-wifi"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                  </div>
                  <div
                    className="col col-md-6"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Parqueadero
                       
                      <i
                        className="fas fa-car"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Baños
                       
                      <i
                        className="fas fa-restroom"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
